sktime/classification/early_classification/_teaser.py

```diff
@@ -0,0 +1,423 @@
+                # 0 if the decision is negative, 1 if its positive but different class
+                else 1 if decisions[i] else 0,
+                # predicted class index
+                preds[i],
+            )
+            for i in range(n_instances)
+        ]
+
+        # return the safety decisions and new state information for the instances
+        return [
+            True if new_state_info[i][1] >= self._consecutive_predictions else False
+            for i in range(n_instances)
+        ], new_state_info
+
+    def _fit_estimator(self, X, y, i):
+        rs = 255 if self.random_state == 0 else self.random_state
+        rs = None if self.random_state is None else rs * 37 * (i + 1)
+        rng = check_random_state(rs)
+
+        estimator = _clone_estimator(
+            WEASEL() if self.estimator is None else self.estimator,
+            rng,
+        )
+
+        # fit estimator for this threshold
+        estimator.fit(X[:, :, : self._classification_points[i]], y)
+
+        m = getattr(estimator, "n_jobs", None)
+        if m is not None:
+            estimator.n_jobs = self._threads_to_use
+
+        # get train set probability estimates for this estimator
+        if callable(getattr(estimator, "_get_train_probs", None)):
+            train_probs = estimator._get_train_probs(X, y)
+        else:
+            cv_size = 5
+            _, counts = np.unique(y, return_counts=True)
+            min_class = np.min(counts)
+            if min_class < cv_size:
+                cv_size = min_class
+
+            train_probs = cross_val_predict(
+                estimator, X, y=y, cv=cv_size, method="predict_proba"
+            )
+
+        rng = check_random_state(self.random_state)
+        train_preds = [
+            int(rng.choice(np.flatnonzero(prob == prob.max()))) for prob in train_probs
+        ]
+        train_probs = np.hstack((train_probs, np.ones((len(X), 1))))
+
+        # create train set for svm using train probs with the minimum difference to the
+        # predicted probability
+        X_svm = []
+        for i in range(len(X)):
+            for n in range(self.n_classes_):
+                if n != train_preds[i]:
+                    train_probs[i][self.n_classes_] = min(
+                        train_probs[i][self.n_classes_],
+                        train_probs[i][train_preds[i]] - train_probs[i][n],
+                    )
+
+            if train_preds[i] == self._class_dictionary[y[i]]:
+                X_svm.append(train_probs[i])
+
+        # tune the one class svm gamma
+        best_svm = None
+        best_acc = -1
+        if len(X_svm) > 1:
+            cv_size = min(len(X_svm), 10)
+
+            for gamma in self._svm_gammas:
+                svm = OneClassSVM(gamma=gamma)
+                threads = 1 if m is None else self._threads_to_use
+                svm_preds = cross_val_predict(svm, X_svm, cv=cv_size, n_jobs=threads)
```

Comment:
Not sure, but isn't this a GridSearchCV?


ocSVM = OneClassSVM(kernel="rbf", tol=1e-4)
parameters = {'gamma':self.SVM_GAMMAS,
                              'nu': self.SVM_NUS}
clf = GridSearchCV(ocSVM, parameters, scoring="accuracy")
clf.fit(confidences, correctPredictions)
self.masterClf[i] = clf.best_estimator_

Reply:
Yeah, that would probably be better. Also saw the tolerance is slightly different to the one you gave, new CV would be:

```
cv_size = min(len(X_svm), 10)
gs = GridSearchCV(
    OneClassSVM(tol=self._svm_tol, nu=self._svm_nu),
    {"gamma": self._svm_gammas},
    scoring="accuracy",
    cv=cv_size,
)
gs.fit(X_svm, np.ones(len(X_svm)))
svm = gs.best_estimator_
```